BE/src/main/java/kr/codesquad/issuetracker/auth/GithubOauth.java

```diff
@@ -0,0 +1,74 @@
+package kr.codesquad.issuetracker.auth;
+
+import static kr.codesquad.issuetracker.auth.utils.OauthUtils.TOKEN;
+
+import kr.codesquad.issuetracker.auth.dto.AccessTokenRequestDto;
+import kr.codesquad.issuetracker.auth.dto.AccessTokenResponseDto;
+import kr.codesquad.issuetracker.auth.dto.UserProfile;
+import kr.codesquad.issuetracker.auth.service.GithubService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+@Component
+public class GithubOauth {
+
+	private final String clientId;
+	private final String clientSecret;
+	private final String accessTokenUri;
+	private final String userUri;
+
+	private final WebClient webClient;
+	private final GithubService githubService;
+
+	public GithubOauth(@Value("${oauth2.user.github.client-id}") String clientId,
+						@Value("${oauth2.user.github.client-secret}") String clientSecret,
+						@Value("${oauth2.provider.github.token-uri}") String accessTokenUri,
+						@Value("${oauth2.provider.github.user-info-uri}") String userUri,
+						WebClient webClient,
+						GithubService githubService) {
+		this.clientId = clientId;
+		this.clientSecret = clientSecret;
+		this.accessTokenUri = accessTokenUri;
+		this.userUri = userUri;
+		this.webClient = webClient;
+		this.githubService = githubService;
+	}
+
+	public AccessTokenResponseDto getToken(String code) {
+		AccessTokenRequestDto accessTokenRequest = AccessTokenRequestDto.builder()
+			.clientId(clientId)
+			.clientSecret(clientSecret)
+			.code(code)
+			.build();
```

Comment:
dto에 빌더패턴은 오버스펙이 아닐까 싶네요.
빌더패턴을 사용하는 이유에 대해서 알아봅시다 :)

생성자가 오히려 간단한 경우가 있습니다.

```suggestion
		AccessTokenRequestDto accessTokenRequest = new AccessTokenRequestDtoclientId, clientSecret, code);
```

Reply:
빌더 패턴을 사용하는 이유중 하나가 생성자 보다 명시적으로 데이터가 어디로 들어가는지 확인가능해서 보기 좋다는 이유로 사용하고있었는데 오버스펙이군요ㅠㅠ 빠르게 수정하겠습니다 감사합니다 :)